lib/ProxySQL_Statistics.cpp

```
@@ -52,9 +103,111 @@ ProxySQL_Statistics::ProxySQL_Statistics() {
+            return true;
+        }
+    } else if (!strcasecmp(name, "monitor_enabled")) {
+        if (intv == 0 || intv == 1) {
+            variables.tsdb_monitor_enabled = intv;
+            return true;
+        }
+    } else if (!strcasecmp(name, "monitor_interval")) {
+        if (intv >= 1 && intv <= 3600) {
+            variables.tsdb_monitor_interval = intv;
+            return true;
+        }
```

Cppcheck (2.19.0)

[error] Common realloc mistake

(memleakOnRealloc)